style: confirm snake_case for files (#958)
tigerbeetle/tigerbeetle: docs/TIGER_STYLE.md

* **Get the nouns and verbs just right.** Great names are the essence of great code, they capture what a thing is or does, and provide a crisp, intuitive mental model. They show that you understand the domain. Take time to find the perfect name, to find nouns and verbs that work together, so that the whole is greater than the sum of its parts.

> * Use `snake_case` for function, variable, and file names. The underscore is the closest thing we have as programmers to a space, and helps to separate words and encourage descriptive names.

Reviewer: Would be great to add:

> Note: We depart from Zig's style here with respect to file names for files containing a type, the rationale being to keep the convention for file names simple and uniform.

Author: PTAL

* **Get the nouns and verbs just right.** Great names are the essence of great code, they capture what a thing is or does, and provide a crisp, intuitive mental model. They show that you understand the domain. Take time to find the perfect name, to find nouns and verbs that work together, so that the whole is greater than the sum of its parts.

> * Use `snake_case` for function, variable, and file names. The underscore is the closest thing we have as programmers to a space, and helps to separate words and encourage descriptive names. We don't use `CamelCase.zig` for "struct" files to keep the convention simple and consistent with our many "generic struct" files.

Reviewer: ```suggestion
* Use `snake_case` for function, variable, and file names. The underscore is the closest thing we have as programmers to a space, and helps to separate words and encourage descriptive names. We don't use Zig's `CamelCase.zig` style for "struct" files to keep the convention simple and consistent.
```
